File: src/_02_StructuralPattern/_02_01_Adapter/java/after/AccountUserDetailService.java
package _02_StructuralPattern._02_01_Adapter.java.after;

import _02_StructuralPattern._02_01_Adapter.java.after.security.UserDetail;
import _02_StructuralPattern._02_01_Adapter.java.after.security.UserDetailService;

public class AccountUserDetailService implements UserDetailService {
  
  private AccountService accountService;

  public AccountUserDetailService(AccountService accountService) {
    this.accountService = accountService;
  }
  
  @Override
  public UserDetail loadUser(String username) {
    return new AccountUserDetail(accountService.findAccountByUsername(username));
  }
  
}

File: src/_02_StructuralPattern/_02_01_Adapter/AccountUserDetailService.js
import AccountUserDetail from "./AccountUserDetail.js";
import UserDetailService from "./UserDetailService.js";

/**
 * @typedef { import("./AccountService").default } AccountService
 * @typedef { import("./UserDetail").default } UserDetail
 */

export default class AccountUserDetailService extends UserDetailService {
  /** @type { AccountService } */
  #accountService;

  /** @param { AccountService } accountService */
  constructor(accountService) {
    super();
    this.#accountService = accountService;
  }

  /**
   * @override
   * @param { string } name
   * @returns { UserDetail }
   */
  loadUser(name) {
    const account = this.#accountService.findAccountByName(name);
    const accountUserDetail = new AccountUserDetail(account);
    return accountUserDetail;
  }
}